purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -372,13 +372,40 @@ class PurchasesTest {
 
     // region get products
 
+    @Test
+    fun getsAllProducts() {
+        val subProductIds = listOf("onemonth_freetrial")
+        val inappProductIds = listOf("normal_purchase")
+        val productIds = subProductIds + inappProductIds
+
+        val subStoreProducts = mockStoreProduct(productIds, subProductIds, ProductType.SUBS)
+        val inappStoreProducts = mockStoreProduct(productIds, inappProductIds, ProductType.INAPP)
+        val storeProducts = subStoreProducts + inappStoreProducts
+
+        purchases.getProducts(productIds,
+            object : GetStoreProductsCallback {
+                override fun onReceived(storeProducts: List<StoreProduct>) {
+                    receivedProducts = storeProducts
+                }
+
+                override fun onError(error: PurchasesError) {
+                    fail("shouldn't be error")
+                }
+            })
+
+        assertThat(receivedProducts).isEqualTo(storeProducts)
+        assertThat(receivedProducts?.size).isEqualTo(productIds.size)
+    }
+
     @Test
     fun getsSubscriptionProducts() {
         val productIds = listOf("onemonth_freetrial")
 
-        val productDetails = mockStoreProduct(productIds, listOf(), ProductType.SUBS)
```

Comment:
I think there was actually a bug here 🤔  The second parameter was returning an empty list of `StoreProduct`s. I've updated this to return a `StoreProduct` and also added a new assert below that checks the size of `productIds` matches the size of `receivedProducts`